### ISSUE TITLE
Feature request: add animation to "::cue" style

### ISSUE THREAD

**Issue by clouds56** [w3c/webvtt#424]
I'd like to add animation to the cue (either css like or [libass](https://github.com/libass/libass) like).

What I'd like to achieve is roll credits like [this](https://forums.adobe.com/thread/2305503), make cue scroll from bottom to top (or from right to left if vertically) until it disappears in the screen.

example:
```
cue1
00:00:01.000 --> 00:00:06.000
line 1
line 2
```
* may use `@keyframes scroll { 0% {top: 100%;} 100% {top: 0%;} }; ::cue{ animation: scroll 5s 1; };` to achieve this.
* when seek the video to some time like 00:00:03, the cue1 should appear at `top: 60%` and scrolling upwards.

**Comment by silviapfeiffer** [comment]
We've got scrolling captions through the region construct. The timing comes from the cues themselves. It's more declarative than the animation-based scrolling you're suggesting. And it's currently limited to scrolling up or down (not vertically left to right or right to left). To cover the whole screen, you'd need to specify more than the standard 3 lines. I think Safari might have implemented regions, if you want to try it out somewhere.

Give it a go and see if that satisfies your needs.

**Comment by clouds56** [comment]
Do you mean the example 1.2 in [this page](https://dvcs.w3.org/hg/text-tracks/raw-file/default/608toVTT/region.html)?
I tried safari, and I think it's actually not scrolling (it's just make some caption jump to avoid overlapping).

**Comment by silviapfeiffer** [comment]
No, that's outdated. Regions are now in the main spec, see https://w3c.github.io/webvtt/#example-cd0199a2 for an example and https://w3c.github.io/webvtt/#regions for the spec.

Implementations indeed still lag behind.

**Comment by dwsinger** [comment]
And indeed we leave it a little open as to how well an implementation does the scrolling, deliberately.

**Comment by Boughtmanatee5** [comment]
I don't know if you've seen the captions with intentions project https://www.captionwithintention.org/. 
Adding animation to `::cue` might make it possible to make these without needing a new caption format.
